_posts/2024-09-05-releasing-apps.md

```diff
@@ -0,0 +1,85 @@
+
+This is the story of our journey towards higher quality, more frequent releases, starting with: Releasing!
+
+--- 
+
+### Slow old GitFlow
+Both our iOS and Android repos used the classic GitFlow branching structure for the longest time. If you’re not familiar with it, here’s a quick recap:
+
+Day-to-day code lands on `develop` over the span of a couple of weeks, after which a  `release` branches is cut. The release candidate is tested and stabilized on the `release` branch before merging to `main` to trigger the deployment. `main` would then need re-merging with `develop`, to ensure those fixes made on `release` made it in to the next one.
+
+![image](/assets/img/2024-09-05-gitflow-branching.png)
+*Dual-trunk branching with GitFlow*
+
+All in all, pretty reasonable! However, it started to become a problem for us as more and more developers started contributing. Branching from a working trunk and stabilizing was fine, as was merging to `main` - a target that isn’t moving. The problem was going the other way - re-integrating `main` with `develop`.
+
+Our working trunk moved quickly, meaning that fixes made on our release branch would likely end up conflicting. Resolving this conflict ended up being a responsibility of the release champion, who sometimes wouldn’t have the context needed to action it correctly without effort. We automated as much as we could, merging this pull request automatically if there wasn’t conflicts. This meant fewer clicks of the “Accept” button, but meant it was easier to miss that a conflict existed and not action the merge at all!
+
+This extra cognitive load on the release champion leads to slower releases. Not only are releases slower and harder to action, with GitFlow we miss out on fixes on the working trunk until the release is finalized. We can mitigate this through _multiple_ pull requests, but that’s even more cognitive load.
+
+### The one trunk to rule them all
+Managing two trunks was starting to get impractical—why not get rid of one of them?
+
+Trunk-based development is an approach where there’s one trunk branch (like `main`), and all work lands there. All work is releasable, made possible by feature flags and automated testing, meaning any given commit is safe to release! While this sounds great in theory, and works well in systems where rolling backwards or forwards is quick and easy, it’s problematic for mobile apps.
+
+Apps are constrained by the stores, not shipping our releases to customers before taking their time reviewing. It takes a good amount longer for adoption to then ramp up, meaning it takes a long time to create a fix and address it. Even if we had incredible test coverage, it’s near-impossible to test every given scenario your app will be run under in production.
+
+A branching structure growing in popularity for mobile apps nowadays is a combination of trunk-based development and release branches. Let’s explain:
```

Comment:
source? I though we came up with this 😅 via evolution

Reply:
Turns out we converged on something others have been doing too: https://www.runway.team/blog/choosing-the-right-branching-strategy-for-mobile-development#trunk-based-w-release-branches-